uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -198,6 +201,10 @@ define([
           canSubmit = canSubmit && !_.isEqual(initialLeaveRequestAttributes, this.request.attributes());
           // user has added a comment
           canSubmit = canSubmit || this.request.comments.length !== initialCommentsLength;
+          // user has added file Attachment
+          canSubmit = canSubmit || (this.request.files.length + this.request.fileUploader.queue.length) !== initialFilesLength;
```

Comment:
After https://github.com/civicrm/civihr/pull/1897 is merged, both `initialFilesLength` and `userRemovedFile` is not necessary anymore.

Instead the following check can be added in `hasRequestChanged`
``` javascript
function hasRequestChanged() {
  // using angular.equals to automatically ignore the $$hashkey property
  return !angular.equals(
    _.omit(initialLeaveRequestAttributes, 'fileUploader'),
    _.omit(this.request.attributes(), 'fileUploader')
  ) || this.request.attributes().fileUploader.queue.length !== 0 ;// this line is added
}
```
So the first condition will take care of any changes to the already uploaded files(deletion) and the or condition will check if new files are added.